src/component/Common/Captcha/CapCaptcha.tsx

```diff
@@ -0,0 +1,99 @@
+import { useEffect, useRef } from "react";
+import { useAppSelector } from "../../../redux/hooks.ts";
+import { CaptchaParams } from "./Captcha.tsx";
+import { Box } from "@mui/material";
+
+export interface CapProps {
+  onStateChange: (state: CaptchaParams) => void;
+  generation: number;
+}
+
+const CapCaptcha = ({ onStateChange, generation, ...rest }: CapProps) => {
+  const captchaRef = useRef<HTMLDivElement>(null);
+  const widgetRef = useRef<any>(null);
+  
+  const capInstanceURL = useAppSelector(
+    (state) => state.siteConfig.basic.config.cap_instance_url,
+  );
+  const capKeyID = useAppSelector(
+    (state) => state.siteConfig.basic.config.cap_key_id,
+  );
+
+  const refreshCaptcha = async () => {
+    if (widgetRef.current) {
```

Comment:
This does not work?

I have this CapCaptcha enabled for sign in, the captcha just disappears after a failed sign in request.